source-kafka/src/main/java/org/apache/kylin/source/kafka/util/KafkaClient.java

```diff
@@ -57,16 +57,16 @@ public static KafkaConsumer getKafkaConsumer(String brokers, String consumerGrou
 
     private static Properties constructDefaultKafkaConsumerProperties(String brokers, String consumerGroup, Properties properties) {
         Properties props = new Properties();
-        if (properties != null) {
-            for (Map.Entry entry : properties.entrySet()) {
-                props.put(entry.getKey(), entry.getValue());
-            }
-        }
         props.put("bootstrap.servers", brokers);
         props.put("key.deserializer", StringDeserializer.class.getName());
         props.put("value.deserializer", StringDeserializer.class.getName());
         props.put("group.id", consumerGroup);
         props.put("enable.auto.commit", "false");
+        if (properties != null) {
```

Comment:
Maybe in terms of design, for these five properties, we just don't want them to be overridden by .xml or .xml.override.